lucille-plugins/lucille-pinecone/src/main/java/com/kmwllc/lucille/pinecone/indexer/PineconeIndexer.java

```diff
@@ -32,49 +31,45 @@
- * - addDeletionPrefix (String, optional) : String that adds on to all document ids used for prefix deletion
- *  - e.g. if id of doc is "doc1" & deletionPrefix is "-", then all vectors containing "doc1-" in id as prefix will be deleted.
  */
 public class PineconeIndexer extends Indexer {
 
   private static final Logger log = LoggerFactory.getLogger(PineconeIndexer.class);
-
+  private static final Integer MAX_PINECONE_BATCH_SIZE = 1000;
   private final Pinecone client;
   private final Index index;
   private final String indexName;
   private final Map<String, Object> namespaces;
   private final Set<String> metadataFields;
   private final String mode;
   private final String defaultEmbeddingField;
-  private final String addDeletionPrefix;
-  private final boolean deleteByPrefix;
 
   public PineconeIndexer(Config config, IndexerMessenger messenger, String metricsPrefix) {
     super(config, messenger, metricsPrefix);
+    this.client = PineconeManager.getClient(config.getString("pinecone.apiKey"));
     this.indexName = config.getString("pinecone.index");
+    this.index = client.getIndexConnection(indexName);
     this.namespaces = config.hasPath("pinecone.namespaces") ? config.getConfig("pinecone.namespaces").root().unwrapped() : null;
     this.metadataFields = config.hasPath("pinecone.metadataFields")
         ? new HashSet<>(config.getStringList("pinecone.metadataFields")) : new HashSet<>();
     this.mode = config.hasPath("pinecone.mode") ? config.getString("pinecone.mode") : "upsert";
     this.defaultEmbeddingField = ConfigUtils.getOrDefault(config, "pinecone.defaultEmbeddingField", null);
-    this.client = new Pinecone.Builder(config.getString("pinecone.apiKey")).build();
-    this.index = this.client.getIndexConnection(this.indexName);
-    this.addDeletionPrefix = config.hasPath("pinecone.addDeletionPrefix") ? config.getString("pinecone.addDeletionPrefix") : "";
-    this.deleteByPrefix = config.hasPath("pinecone.deleteByPrefix") ? config.getBoolean("pinecone.deleteByPrefix") : false;
     if (namespaces != null && namespaces.isEmpty()) {
-      throw new IllegalArgumentException("namespaces mapping must be non-empty if provided");
+      throw new IllegalArgumentException("Namespaces mapping must be non-empty if provided.");
     }
-    // max upsertSize is 2MB or 1000 records, whichever is reached first, so stopping lucille run if batch size set to more than 1000
+    // max upload batch is 2MB or 1000 records, whichever is reached first, so stopping lucille run if batch size set to more than 1000
     // larger dimensions will mean smaller batch size limit, letting API throw the error if encountered.
-    if (mode.equalsIgnoreCase("upsert") && DEFAULT_BATCH_SIZE > 1000) {
-      throw new IllegalArgumentException("maximum batch size for upsert is 1000, and lower if vectors have higher dimensions");
+    if (DEFAULT_BATCH_SIZE > MAX_PINECONE_BATCH_SIZE) {
+      log.warn(
```

Comment:
as discussed, this should check what the batch size is actually set to (which could happen via indexer.batchSize or via the default) and log the warning only if it's too high

Reply:
Ticket [created](https://kmwllc.atlassian.net/browse/LC-554?atlOrigin=eyJpIjoiOWYxYzM1ZjZkNGZlNDJlMGI5NGZlMDY2YTQ2NjVlNWMiLCJwIjoiaiJ9) to enable retrieval of indexer batch size.

Reply:
replaced in [f745a9e](https://github.com/kmwtechnology/lucille/pull/187/commits/f745a9e5b69d49fdd8b91e8728ea27d4dc3406de) to check for actual batch size